### DISCUSSION TITLE
[Feature] Multiple Storage Tiers with Automated Archiving

### DISCUSSION THREAD

**Discussion by lohrbini** [immich-app/immich#4620] [Feature Request]
### Feature Description:
We propose the introduction of multiple storage tiers, which would enable us to segregate data into different storage classes based on its frequency of access, age or storage templates. These tiers could be:

- local storage (ssd/nvme)
- remote storage (s3)

### Key Benefits:
Cost-Efficiency: 

- Storing rarely accessed data in the Archive Storage Tier would significantly reduce primary storage costs, as object storage solutions are generally more cost-effective for long-term data retention.

Data Preservation: 

- Archiving older data ensures that we maintain a complete historical record while freeing up primary storage for new and active data.

### Configuration Options:
Allow users to configure the criteria for data migration to the Archive Storage Tier, such as access frequency, file age, or custom tags. Additionally, provide options for data restoration from the Archive Storage Tier when needed.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by AngelaDMerkel** [comment]
Would all of the thumbnails be stored locally too? I assume so.

**Comment by akoyaxd** [comment]
I think a good example of impelmenting something similar is the matrix media repository from Travis Ralston:
https://docs.t2bot.io/matrix-media-repo/v1.3.3/configuration/datastores.html

Basically having multiple Datastores of either local filesystem or s3. So a user could decide where he wants to store which kind of asset (Thumbnail, original, video-preview etc.)

E.g. store thumbnails in local storage or maybe use a fast s3 provider and store assets in a slower, cheaper s3 tier.

**Comment by LM1LC3N7** [comment]
It would be great if Immich could move image from a "hot" (speed) default storage for newer files to a "warm" or "cold" storage for older files, **not just a s3** location. 

Example of what I imagine:

Files newer that 6 month or a year (configurable) are stored by default on "LOCATION_HOT" which is a SSD on my server
Filer older that this limit are moved to another location, which could be my 12Tb disk
This would save disk storage for expensive hardware, while maintaining a good user experience through the app. And also:

Memories of the day (or the month in order to batch operations) are copied or moved temporarily to the HOT storage, in case the user wants to download them or view the original file, not just the thumbnail
All cache and thumbnails is by default on the HOT storage
For small hardware and servers, the location of HOT and COLD is the same. But for advanced users, it can be a local SSD and a local HDD, or even a local disk and a distant one (or S3).

**Comment by Jarod1230** [comment]
Any new updates on that? Would still be a very useful feature :)

**Reply by bo0tzz** [reply]
https://justinmayer.com/posts/any-updates/

**Reply by jagjordi** [reply]
it would still be nice to get a word on the status for example: not going to happen, might happen in ~1 year, or actively working on it. Doesn't have to be very precise

**Reply by bo0tzz** [reply]
The request is open meaning we haven't ruled it out, and there's no plans stated here or on the roadmap meaning there are no plans to work on it.
